functions/node/src/index.ts

```diff
@@ -0,0 +1,25 @@
+import { initializeApp } from 'firebase-admin/app';
+import { getFirestore } from 'firebase-admin/firestore';
+import { beforeUserCreated, beforeUserSignedIn, HttpsError, type AuthBlockingEvent } from 'firebase-functions/identity';
+
+initializeApp();
+
+const db = getFirestore();
+
+async function checkUser(event: AuthBlockingEvent) {
+  const email = event.data?.email;
+
+  if (!email) {
+    throw new HttpsError('invalid-argument', 'Email is required');
+  }
+
+  const ref = db.collection('authorized-users').doc(email);
+  const doc = await ref.get();
+
+  if (!doc.exists) {
+    throw new HttpsError('not-found', `User (${email}) is not authorized`);
+  }
+}
+
+export const beforeCreated = beforeUserCreated(checkUser);
+export const beforeSignedIn = beforeUserSignedIn(checkUser);
```

Comment:
Would you share your learning that helped you decide that the function is needed in both places?

Reply:
I wasn't sure. I do know that created fires before signed in. So the only situation where I could think that it would be necessary is if there were already existing users or we wanted to enhance the check to check additional user properties. I can remove the signed in if you would prefer.

Reply:
I'm good with both. How about adding some comments to explain that a little.